kernel-rs/src/sysfile.rs

```diff
@@ -304,11 +286,15 @@ pub unsafe fn sys_open() -> usize {
 
     if ip.typ == T_DEVICE {
         (*f).typ = FileType::Device {
-            ip: ptr,
+            //TODO : Use better code
+            ip: *(&ip.ptr as *const _ as *mut _),
```

Comment:
```suggestion
            ip: ip.ptr as *const _ as *mut _,
```
그리고 이 함수 위의 `clippy::allow(..)` 를 제거할 수 있습니다.

Reply:
https://github.com/kaist-cp/rv6/pull/228/commits/2aa14b4546edfa2a07160b0b95d7e3a745b3eff2
https://github.com/kaist-cp/rv6/pull/228/commits/8e8875f320cf64b909cc98933d11de7364fa8007